lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -524,4 +524,113 @@ public void resume_verify_publish() {
 				ablyRx.close();
 		}
 	}
+
+	/**
+	 * Connect to the service using two library instances to set
+	 * up separate send and recv connections.
+	 *
+	 * Send some messages, drop the sender's transport, then send another
+	 * round of messages which should be queued and published after
+	 * we reconnect the sender.
+	 */
+	@Test
+	public void resume_publish_queue() {
+		AblyRealtime receiver = null;
+		AblyRealtime sender = null;
+		String channelName = "resume_publish_queue";
+		int messageCount = 3;
+		long delay = 200;
+		try {
+			ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+			receiver = new AblyRealtime(opts);
+			sender = new AblyRealtime(opts);
+
+			/* create and attach channel to send on */
+			final Channel senderChannel = sender.channels.get(channelName);
+			senderChannel.attach();
+			(new ChannelWaiter(senderChannel)).waitFor(ChannelState.attached);
+			assertEquals(
+			    "The sender's channel should be attached!",
```

Comment:
Just as a minor point of style: we don't need the `!`. Simple, factual, undramatic statements are fine :)